File: solutions/258.add-digits/add-digits.js
/**
 * @param {number} num
 * @return {number}
 */
const addDigits = num => {
  if (num === 0) return 0
  if (num > 0 && num < 10) {
    return num
  } else {
    let sum = 0
    while (num >= 1) {
      sum += num % 10
      num = Math.floor(num / 10)
    }

    return addDigits(sum)
  }
}

File: solutions/258.add-digits/add-digits.py
class Solution:
    def addDigits(self, num):
        """
        :type num: int
        :rtype: int
        """
        while num > 9:
            s = str(num)
            next = 0

            for c in s:
                next += int(c)

            num = next

        return num
